code/modules/clothing/head/hardhat.dm

```diff
@@ -3,45 +3,41 @@
 	desc = "A piece of headgear used in dangerous working conditions to protect the head. Comes with a built-in flashlight."
 	icon_state = "hardhat0_yellow"
 	item_state = "hardhat0_yellow"
-	var/brightness_on = 4 //luminosity when on
-	var/on = 0
-	var/hardhat_color = "yellow" //Determines used sprites: hardhat[on]_[hardhat_color]
 	soft_armor = list("melee" = 30, "bullet" = 5, "laser" = 20, "energy" = 10, "bomb" = 20, "bio" = 10, "rad" = 20, "fire" = 10, "acid" = 10)
 	actions_types = list(/datum/action/item_action/toggle)
 	siemens_coefficient = 0.9
 	flags_inventory = BLOCKSHARPOBJ
+	light_system = MOVABLE_LIGHT
+	light_on = FALSE
+	light_range = 4
+	light_power = 2
+	var/hardhat_color = "yellow" //Determines used sprites: hardhat[on]_[hardhat_color]
 
 /obj/item/clothing/head/hardhat/attack_self(mob/user)
 	if(!isturf(user.loc))
 		to_chat(user, "You cannot turn the light on while in [user.loc]")
 		return
-	toggle_helmet_light()
-	icon_state = "hardhat[on]_[hardhat_color]"
-	item_state = "hardhat[on]_[hardhat_color]"
+	toggle_light(user)
+
+/obj/item/clothing/head/hardhat/proc/toggle_light(mob/user)
+	turn_light(user, !light_on)
 
+/obj/item/clothing/head/hardhat/turn_light(mob/user, toggle_on)
```

Comment:
is there a particular reason this is a wrapper?

Reply:
since it's called by another object, we don't really have access to light_on, and it's cleaner to have a wrapper. I could do without it, though